ckanext/datastore/blueprint.py

```diff
@@ -163,63 +206,103 @@ def post(self, id: str, resource_id: str):
 
 
 def dump_to(
-    resource_id: str, output: Any, fmt: str, offset: int, limit: Optional[int],
-        options: dict[str, Any], sort: str, search_params: dict[str, Any]
+    resource_id: str, output_stream: Any, fmt: str, offset: int,
+    limit: Optional[int], options: dict[str, Any], sort: str,
+    search_params: dict[str, Any], user: str, is_generator: bool = True
 ):
-    if fmt == u'csv':
+    if fmt == 'csv':
         writer_factory = csv_writer
-        records_format = u'csv'
-    elif fmt == u'tsv':
+        records_format = 'csv'
+    elif fmt == 'tsv':
         writer_factory = tsv_writer
-        records_format = u'tsv'
-    elif fmt == u'json':
+        records_format = 'tsv'
+    elif fmt == 'json':
         writer_factory = json_writer
-        records_format = u'lists'
-    elif fmt == u'xml':
+        records_format = 'lists'
+    elif fmt == 'xml':
         writer_factory = xml_writer
-        records_format = u'objects'
+        records_format = 'objects'
     else:
-        assert False, u'Unsupported format'
+        assert False, 'Unsupported format'
+
+    bom = options.get('bom', False)
 
-    def start_writer(fields: Any):
-        bom = options.get(u'bom', False)
-        return writer_factory(output, fields, resource_id, bom)
+    def start_stream_writer(output_stream: StringIO,
+                            fields: list[dict[str, Any]]):
+        return writer_factory(output_stream, fields, bom=bom)
 
-    def result_page(offs: int, lim: Optional[int]):
-        return get_action(u'datastore_search')(
-            {},
+    def stream_result_page(offs: int, lim: Union[None, int]):
+        return get_action('datastore_search')(
+            {'user': user},
             dict({
-                u'resource_id': resource_id,
-                u'limit': PAGINATE_BY
-                if lim is None else min(PAGINATE_BY, lim),
-                u'offset': offs,
-                u'sort': sort,
-                u'records_format': records_format,
-                u'include_total': False,
+                'resource_id': resource_id,
+                'limit': PAGINATE_BY
+                if limit is None else min(PAGINATE_BY, lim),  # type: ignore
+                'offset': offs,
+                'sort': sort,
+                'records_format': records_format,
+                'include_total': False,
             }, **search_params)
         )
 
-    result = result_page(offset, limit)
+    def stream_dump(offset: int, limit: Union[None, int],
+                    paginate_by: int, result: dict[str, Any]):
+        with start_stream_writer(output_stream, result['fields']) as output:
+            while True:
+                if limit is not None and limit <= 0:
+                    break
+
+                records = result['records']
+
+                output.write_records(records)
+                output_stream.seek(0)
```

Comment:
streams are by definition file-like objects that you can't call seek on, so this should be renamed `output_buffer` or similar.

Since all the users of dump_to either create a StringIO to pass as this variable or pass a file with `is_generator=False` I think it's better to remove both parameters, create the StringIO object inside the function and always use it as a generator.

You can stream the `dump_to` generator to a file with
```python
for block in dump_to(...):
    myfile.write(block)
```
instead of passing in myfile and is_generator=False

Reply:
@wardi okay cool, yeah I did not know how to get a generator to write to a file. This has been done now and works